presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2025,6 +2025,28 @@ public PartitionStatistics getStatisticsUpdate()
             return statisticsUpdate;
         }
 
+        public Table getAugmentedTableForInTransactionRead()
+        {
+            // Don't augment the location for partitioned tables,
+            // as data is never read directly from the partitioned table location
+            if (!table.getPartitionColumns().isEmpty()) {
+                return table;
+            }
+
+            // For unpartitioned table, this method augments the location field of the table
+            // to the staging location.
+            // This way, if the table is accessed in an ongoing transaction, staged data
+            // can be found and accessed.
+            Table table = this.table;
```

Comment:
I know that it is implemented that way for partition `Table table = this.table`, and maybe it makes some sense to keep it so for consistency, but frankly the `Table table = this.table` looks weird. Why not just use the `this.table` as `table` directly. Also why not to short cirtcut for all the corner cases the same way as it is done for `!table.getPartitionColumns().isEmpty()`, and then leave the

```
return Table.builder(table)
                        .withStorage(storage -> storage.setLocation(currentLocation))
                        .build();
```

As the last statement